runner/log/docker.go

```diff
@@ -49,18 +49,20 @@ func (d Docker) Run() op.Op {
 			runner.Params(d))
 	}
 
+	// Check whether the container can be found on the system
+	if !d.containerExists() {
+		return op.New(d.ID(), "", op.Skip, ContainerNotFoundError{
+			container: d.Container,
+		},
+			runner.Params(d))
+	}
+
 	// Retrieve logs
 	cmd := DockerLogCmd(d.Container, d.DestDir, d.Since)
 	o = runner.NewSheller(cmd, d.Redactions).Run()
-	// NOTE(mkcp): If the container does not exist, docker will exit non-zero and it'll surface as a ShellExecError.
-	//  The result actionably states that the container wasn't found. In the future we may want to scrub the result
-	//  and only return an actionable error message
 	if o.Error != nil {
 		return op.New(d.ID(), o.Result, o.Status, o.Error, runner.Params(d))
 	}
-	if o.Result == "" {
```

Comment:
Because we redirect the shell output, it appeared in testing that the result was always `""`. This meant that we could gather logs but still warn that there were no logs. I removed this check and warning as a result of this behavior, so we wouldn't end up with conflicting information.